entitlement/src/main/java/com/hevelian/identity/entitlement/engine/EntitlementEngine.java

```diff
@@ -0,0 +1,149 @@
+package com.hevelian.identity.entitlement.engine;
+
+import com.hevelian.identity.entitlement.ctx.RequestCtxFactory2;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.Setter;
+import org.w3c.dom.Element;
+import org.wso2.balana.PDP;
+import org.wso2.balana.PDPConfig;
+import org.wso2.balana.ParsingException;
+import org.wso2.balana.ctx.RequestCtxFactory;
+import org.wso2.balana.ctx.ResponseCtx;
+import org.wso2.balana.ctx.xacml3.RequestCtx;
+import org.wso2.balana.finder.AttributeFinder;
+import org.wso2.balana.finder.AttributeFinderModule;
+import org.wso2.balana.finder.PolicyFinder;
+import org.wso2.balana.finder.ResourceFinder;
+import org.wso2.balana.finder.impl.CurrentEnvModule;
+import org.wso2.balana.finder.impl.SelectorModule;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents a policy engine, providing the point for request evaluation.
+ */
+public class EntitlementEngine {
+  @Getter
+  @Setter(AccessLevel.PROTECTED)
+  private PDP pdp;
+
+  /**
+   * Initializes {@link PDP} instance.
+   * <p>
+   * For correct work the 'evaluate' functions this object must be initialized, otherwise you will get {@link NullPointerException}.
```

Comment:
For a correct work of "evaluate" functions - this object must be initialized, otherwise...